dll/cpl/hotplug/eject.c

```diff
@@ -0,0 +1,130 @@
+/*
+ * PROJECT:     ReactOS Safely Remove Hardware Applet
+ * LICENSE:     GPL-2.0-or-later (https://spdx.org/licenses/GPL-2.0-or-later)
+ * PURPOSE:     Device Removal Support
+ * COPYRIGHT:   Copyright 2023 Thamatip Chitpong <thamatip.chitpong@reactos.org>
+ */
+
+#include "hotplug.h"
+
+#include <strsafe.h>
+
+DEVINST
+GetDeviceInstForRemoval(
+    _In_ PHOTPLUG_DATA pHotplugData)
+{
+    HTREEITEM hTreeItem;
+    TVITEMW tvItem;
+
+    hTreeItem = TreeView_GetSelection(pHotplugData->hwndDeviceTree);
+    if (!hTreeItem)
+        return 0;
+
+    /* Find top-level parent item */
+    while (TreeView_GetParent(pHotplugData->hwndDeviceTree, hTreeItem))
```

Comment:
Two questions:
- is that top-level parent (one of) the root(s) in the treeview?
- It looks the loop could slightly be simplified (at the expense of introducing another TVITEM variable) in order to not call twice in successive manners TreeView_GetParent with the very same parameters.

Reply:
> is that top-level parent (one of) the root(s) in the treeview?

Yes.

> It looks the loop could slightly be simplified (at the expense of introducing another TVITEM variable) in order to not call twice in successive manners TreeView_GetParent with the very same parameters.

Improved.